src/tool/xmeta/cpp/models/enum_model.h

```diff
@@ -46,11 +46,26 @@ namespace xlang::xmeta
             return m_value;
         }
 
-        void set_value(enum_value_semantics const& value)
+
+        void set_value(model_ref<enum_value_semantics> const& value)
         {
             m_value = value;
         }
 
+        enum_value_semantics get_resolved_value() const
+        {
+            assert(m_value.is_resolved());
+            return std::visit(
+                [&](auto&& value) -> enum_value_semantics
+                {
+                    if constexpr (!std::is_integral_v<std::decay_t<decltype(value)>>)
```

Comment:
@DefaultRyan Won't this always be an integral type? `m_value.get_resolved_target()` is a variant of only integral types.

Reply:
That's correct. I think we can get away with:
```c++
static_assert(std::is_integral_v<std::decay_t<decltype(value)>>);
return value;
```